orchestra/cmds/shell.py

```diff
@@ -49,14 +50,21 @@ def handle_shell(args):
 
     if command:
         script_to_run = " ".join(shlex.quote(c) for c in command)
-        p = run_script(script_to_run, environment=env, strict_flags=False, check_returncode=False, cwd=cd_to)
-        exit(p.returncode)
+        p = _run_script(
+            script_to_run,
+            environment=env,
+            strict_flags=False,
+            cwd=cd_to,
+            loglevel="DEBUG"
+        )
+        return p.returncode
+
+    user_shell = get_script_output("getent passwd $(whoami) | cut -d: -f7").strip()
 
-    user_shell = run_script("getent passwd $(whoami) | cut -d: -f7", quiet=True).stdout.decode("utf-8").strip()
     env["OLD_HOME"] = os.environ["HOME"]
     env["HOME"] = os.path.join(os.path.dirname(__file__), "..", "support", "shell-home")
     env["PS1_PREFIX"] = ps1_prefix
-    script = dedent(f"""
-    {user_shell}
-    """)
-    run_script(script, environment=env, cwd=cd_to)
+    script = dedent(f"exec {user_shell}")
+
+    result = _run_script(script, environment=env, loglevel="DEBUG", cwd=cd_to)
+    return result.returncode
```

Comment:
Why are we using a private function? Can't we use `run_internal_script`?

Reply:
It's to override the loglevel so the user is not spammed with "starting /bin/bash".
`orc shell` is a bit special and we used to just use subprocess directly, which I still think might be appropriate since we don't need anything the `run_script` helpers do.